torch/csrc/utils/future.h

```diff
@@ -47,18 +47,36 @@ class TORCH_API Future final {
       completed_ = true;
       value_ = std::move(value);
 
-      fireCallbacks();
+      std::vector<Callback> cbs;
+      cbs.swap(callbacks_);
```

Comment:
Can you add a comment why you're moving the callbacks here? This will be useful for the next person reading this code that doesn't know why this happens (e.g. "Move callbacks to a vector on the stack so we can access it without holding a lock").